test_suite/tests/test_annotations.rs

```diff
@@ -811,6 +811,77 @@ fn test_serialize_with_enum() {
     );
 }
 
+#[derive(Debug, PartialEq, Serialize)]
+enum SerializeWithVariant {
+    #[serde(serialize_with="serialize_unit_variant_as_i8")]
+    Unit,
+
+    #[serde(serialize_with="SerializeWith::serialize_with")]
+    Newtype(i32),
+
+    #[serde(serialize_with="serialize_some_other_variant")]
+    Tuple(String, #[serde(skip_serializing_if="is_zero")] u8),
+
+    #[serde(serialize_with="serialize_some_other_variant")]
+    Struct {
+        f1: String,
+        #[serde(skip_serializing_if="is_zero")]
```

Comment:
I would prefer not to support skip attributes inside of a variant with serialize_with until somebody has a compelling use case for it. For now the derive should panic so it fails to compile. I feel like it is just as convenient and more obvious for this logic to be encompassed inside the serialize_with function:

```rust
fn serialize_some_other_variant<S>(f1: &String,
                                   f2: &u8,
                                   serializer: S)
                                   -> Result<S::Ok, S::Error>
    where S: Serializer
{
    if is_zero(*f2) {
        /* ... */
    } else {
        /* ... */
    }
}
```